py-polars/polars/dataframe/frame.py

```diff
@@ -3834,6 +3835,32 @@ def unpack_table_name(name: str) -> tuple[str | None, str | None, str]:
             msg = f"unrecognised connection type {connection!r}"
             raise TypeError(msg)
 
+    def write_iceberg(
+        self,
+        table: pyiceberg.table.Table,
```

Comment:
Why does this method take a Table and not a Pathlike, like other write functions?

Reply:
Writing to an iceberg table requires creating the `pyiceberg.table.Table` object, which also requires an iceberg catalog. 
I think it's best to delegate the creation of the catalog and table to be outside of this function and just pass in the table object. 


The original implementation takes a str/Pathlike representing the "warehouse location" for the iceberg table. In Iceberg world, writing to a table usually requires an update to the catalog.